server/pulp/plugins/util/misc.py

```diff
@@ -40,13 +40,16 @@ def mkdir(*args, **kwargs):
     """
     Create the specified directory.
     Tolerant of race conditions.
+    Sets umask to 002.
 
     :param args: path[, mode] that goes to os.makedirs
     :param kwargs: path
     :return:
     """
     try:
+        mask = os.umask(0002)
         os.makedirs(*args, **kwargs)
```

Comment:
The umask is apparently never restored if this raises an exception.  Should the original os.umask be above the "try", and the umask reset in a new "finally" block?

Reply:
...I am an idiot. I had exactly that at some point, and apparently lost it somewhere along the way. Thank you for the great catch, that would have been *very* embarrassing to allow out into the wild!

Reply:
@rohanpm @ipanova fixed mkdir() - more eyes please, when you have a moment free